mlflow/tracing/trace_manager.py

```diff
@@ -56,13 +57,15 @@ def __init__(self):
             maxsize=MLFLOW_TRACE_BUFFER_MAX_SIZE.get(),
             ttl=MLFLOW_TRACE_BUFFER_TTL_SECONDS.get(),
         )
+        # Store mapping between OpenTelemetry trace ID and MLflow request ID
+        self._trace_id_to_request_id = {}
```

Comment:
note: Will add per-key lock logic in the follow-up PR with actually calling `StartTrace` API.

Reply:
Seems in the new logic using trace_id is much simpler than request_id for maintaining the traces